playbooks/adhoc/docker_storage_to_overlay2/docker_storage_to_overlay2.yml

```diff
@@ -0,0 +1,197 @@
+#
+#    https://github.com/moby/moby/pull/24771
+#
+#    and
+#
+#    https://github.com/moby/moby/pull/32977
+#
+# Note, this makes no attempt to migrate existing containers/images.
+# It momentarily stops docker.service and uses "atomic storage reset"
+# to wipe all local Docker storage.
+#
+# To run:
+#
+# ansible-playbook -e 'cli_tag_name=<tag-name>' docker_storage_to_overlay2.yml
+#
+
+- name: Transition Docker storage driver to overlay2
+  hosts:
+  - masters
+  - nodes
+  user: root
+  connection: ssh
+
+  # For 'ansible_distribution*'
+  gather_facts: yes
+
+  vars:
+    storage_lvname: "docker-root-lv"
+    storage_lvsize: "100%FREE"
+    storage_setup_path: "/etc/sysconfig/docker-storage-setup"
+
+    # List of services to stop and restart.  Any available services
+    # in optional_service_names get appended to this list beforehand.
+    service_names:
+    - "docker.service"
+
+    optional_service_names:
+    - "atomic-openshift-node.service"
+
+    package_requirements:
+
+      # Ensure we have a fix for the --lvsize option; previously it would
+      # try to call into /usr/lib/docker-storage-setup/libdss.sh, instead
+      # of /usr/share/container-storage-setup/libcss.sh.
+      atomic:
+        minimum: 1.19.1-3
+        reason: "a 'storage modify' bugfix"
+
+      # Ensure XFS volumes are mounted with the pquota option.
+      container-storage-setup:
+        minimum: 0.7.0-1
+        reason: "pquota mount option"
+
+      # Ensure we have the backported overlay2.size storage option.
+      docker:
+        minimum: 1.12.6-58
+        reason: "overlay2.size storage option"
+
+  tasks:
+
+  - name: Install or update required packages
+    package:
+      name: "{{ item.key }}"
+      state: latest
+    with_dict: "{{ package_requirements }}"
+
+  - name: Start docker.service
+    service:
+      name: docker
+      state: started
+
+  - name: Capture output of "docker info"
+    command: docker info
+    register: docker_info_output
+    changed_when: False
+    ignore_errors: yes
+
+  - name: Isolate storage driver
+    set_fact:
+      old_storage_driver: "{{ docker_info_output.stdout | regex_search('(?<=Storage Driver:)\\s*(\\w+)', '\\1', ignorecase=True, multiline=True) }}"
+
+  - name: Transition storage driver to overlay2
+    when: old_storage_driver[0] != "overlay2"
+    block:
+
+    # As of 7.4, SELinux security labels are now supported
+    # on the OverlayFS file system (see RHEL release notes).
+    - name: Check if distribution supports overlayfs (RHEL/CentOS)
+      when: ansible_distribution in ["RedHat", "CentOS"]
+      assert:
+        that: ansible_distribution_version | version_compare("7.4", "ge")
+        msg: "The OverlayFS file system only works with SELinux in {{ ansible_distribution }} 7.4 or later."
+
+    # XXX Is there no easier way to do this?
+    - name: Get {{ item.key }} version
+      command: rpm --query --queryformat="%{version}-%{release}" "{{ item.key }}"
+      args:
```

Comment:
We have a module we prefer to use for this, `repoquery`. For example:

```
- name: Get {{ item.key }} version
  repoquery:
    name: "{{ item.key }}"
    query_type: installed
    show_duplicates: True
  register: package_version
```

And I believe the data you want would be in `package_check_out.results.versions.available_versions_full`.

Reply:
(This is more of a "nice to have" feature, btw. It doesn't have to hold up the stg testing.)